tests/test_type_inference/utils.py

```diff
@@ -0,0 +1,5 @@
+def eq_(a, b, msg=None):
```

Comment:
You have the right instinct for preserving this functionality, but this isn't necessary for our case. It turns out `pytest` itself provides good error messages when assertions fail, and we should rely on that.

So please remove this, and in the test files above, replace `eq_(a, b)` with `assert a == b` and `eq_(a, b, msg)` with `assert a == b, msg`.